src/annotation/target_name.ts

```diff
@@ -3,7 +3,7 @@ import { Metadata } from '../planning/metadata';
 import { tagParameter, DecoratorTarget } from './decorator_utils';
 
 function targetName(name: string) {
-  return function (target: DecoratorTarget, targetKey: string, index: number) {
+  return function (target: DecoratorTarget, targetKey: string | undefined, index: number) {
```

Comment:
Could even be this, and would still work

```suggestion
  return function (target: DecoratorTarget, targetKey: undefined, index: number) {
```